src/Swift.Runtime/src/Metadata/ValueWitnessTable.cs

```diff
@@ -101,6 +101,24 @@ public unsafe ref struct ValueWitnessTable
 		/// </summary>
 		public uint ExtraInhabitantCount;
 
+		/// <summary>
+		/// Returns a tag for the given enumeration. The tags start at 0 with the first
+		/// case with a payload and continue in declaration order. Then they are followed by
+		/// cases without payloads in declaration order.
+		/// </summary>
+		public delegate* unmanaged<void *, TypeMetadata, nuint> GetEnumTag;
```

Comment:
Nit. I would move these definitions a few lines above after https://github.com/dotnet/runtimelab/blob/64b5b7f6e9ddee65a74cbc00f3139b5a757f7ccd/src/Swift.Runtime/src/Metadata/ValueWitnessTable.cs#L82 to keep it at one place.

Reply:
I can't do that, though. The order is significant since we are taking these directly from memory.

Reply:
Ah, good point, I haven't realized that. I think we should explicitly mention that we need to keep the order in sync with Swift def file.

Another question... what mechanism will we use to distinguish what parts should be used for which types. Since these 3 added APIs are for Enum only, we need to make sure we are not trying to read them for other types as well.